.gitignore

```diff
@@ -100,3 +100,8 @@ python-wheels/
 
 # python wheels
 python/share/*
+
+# git merge
+*.orig
+\#*
+\.#*
```

Comment:
@fengleitian You made a comment about the whitespace here, is there an issue?

Reply:
I think clang build with warning as error might fail due to this.

Reply:
I was wrong, for this .gitignore file, here is the reason:
https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

Reply:
For a long time, many UNIX programs did not see the last line of a file if it did not have a new line at the end of it, which is why this check exists.